04_setup_ironic.sh

```diff
@@ -75,6 +55,12 @@ sudo podman run -d --net host --privileged --name ironic --pod ironic-pod \
      --env MARIADB_PASSWORD=$mariadb_password \
      -v $IRONIC_DATA_DIR:/shared ${IRONIC_IMAGE}
 
+sudo podman run -d --net host --privileged --name ipa-downloader --pod ironic-pod \
+     -v $IRONIC_DATA_DIR:/shared ${IPA_DOWNLOADER_IMAGE} /usr/local/bin/get-resource.sh
+
+sudo podman run -d --net host --privileged --name coreos-downloader --pod ironic-pod \
+     -v $IRONIC_DATA_DIR:/shared ${COREOS_DOWNLOADER_IMAGE} /usr/local/bin/get-resource.sh $RHCOS_IMAGE_URL
+
```

Comment:
In https://github.com/openshift-metal3/kni-installer/pull/98 @stbenjam is adding logic to wait for the ironic API to be ready, but the deployment will fail if the image isn't downloaded, so I guess we should poll for the images here, to be sure they're ready before we start ironic?

Although the other question is how/where we do that when this is running on the bootstrap VM, since that will just be a podman pod start, so perhaps we need to check image availability somewhere in kni-installer instead?

What are your thoughts @stbenjam ?

Reply:
Alternatively maybe we should add some logic to the runironic script, so we don't start the API until the expected images are available?

Reply:
Not bringing up Ironic until the images are available sounds like the easiest thing to do - either in runironic or in whatever is launching the Ironic pods.

Terraform could also have a check for the images being available before trying to provision the masters

Reply:
This should "just work" on kubernetes since this can be an 'init' container.  Then it runs before the rest of the pod comes up.  I'll try to test this in kubernetes this week.

Reply:
I looked at this when running on the bootstrap VM and it's tricky because sometimes the images take a long time to download (on one occasion it took 17minutes to download a non-cached image!) and we've got a 10 minute timeout on the systemd service.

We could increase that, but I'm wondering if it would be clearer to just wait in terraform, so the user gets clear feedback about why/where we're waiting?